sql/core/src/main/scala/org/apache/spark/sql/parquet/newParquet.scala

```diff
@@ -465,7 +498,7 @@ case class ParquetRelation2
     // before calling execute().
 
     val job = new Job(sqlContext.sparkContext.hadoopConfiguration)
-    val writeSupport = if (schema.map(_.dataType).forall(_.isPrimitive)) {
+    val writeSupport = if (parquetSchema.map(_.dataType).forall(_.isPrimitive)) {
```

Comment:
What will happen if we do `save` with `SaveMode.Overwrite` and the given `DataFrame` has a different schema with the existing one?

Should we use the schema of `data`?